File: day6/Python/wordRev.py
"""
  * @author : ashwek
  * @date : 27/12/2018
"""

def rev(Str, start, end):

    for i in range(0, (end-start)//2):
        Str[i+start], Str[end-i-1] = Str[end-i-1], Str[i+start]

def wordRev(Str):

    Str = list(Str)
    start = 0

    while( True ):

        try: end = Str.index(' ', start)
        except ValueError: break

        rev(Str, start, end)
        start = end+1

    rev(Str, start, len(Str))

    return ''.join(Str)


Str = input("Enter a string = ")

print("Words Reversed =", wordRev(Str))

File: day6/Java/WordRev.java
/**
 * @author MadhavBahl
 * @date 27/12/2018
 */

import java.util.Scanner;
import java.lang.*;

public class WordRev {
    public static void main(String[] args) {
        Scanner input = new Scanner (System.in);
        System.out.print("Enter the sentence: ");
        String sentence = input.nextLine();

        String[] words = sentence.split("\\s+");
        String reversed;

        for (int i=0; i<words.length; i++) {
            reversed = "";
            for (int j=0; j<words[i].length(); j++) {
                reversed = words[i].charAt(j) + reversed;
            }
            words[i] = reversed;
        }

        String wordsReversed = String.join(" ", words);
        System.out.println("String with reversed words: " + wordsReversed);
    }
}
